src/bootstrap-table.js

```diff
@@ -602,7 +602,7 @@
         } else if (type === 'prepend') {
             this.data = [].concat(data).concat(this.data);
         } else {
-            this.data = data || this.options.data;
+            this.data = this.options.data || data;
```

Comment:
this.data = data || this.options.data;

Currently the sorting of data on refresh is always done against "data" (new data) that has not been processed by the flatJSON extension.

this.data (on line 622) in BootstrapTable.prototype.initSort depends on the chosen data on line 605. If this.data is not flattened yet, the sorting could be done against a data-field with dot notation that does not exist in this.data. Which ends up with the selected sorting lost on refresh.

    this.data = this.options.data (new flattened data from flatJSON) || data (new data)